File: springbootjparest/src/main/java/com/scarc/springbootjparest/models/TimeSlot.java
package com.scarc.springbootjparest.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalTime;
import java.util.List;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class TimeSlot {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;
  private LocalTime starts;
  private LocalTime ends;
  private int maxParticipants;

  @ManyToOne
  private Participation participation;

  @ManyToMany
  private List<Pupil> pupils;
}

File: KOTLIN/Wirtschaftstag/Kotlin_Wirtschaftstag/androidApp/src/main/java/me/scarc/androidApp/data/TimeSlot.kt
package me.scarc.androidApp.data

import java.time.LocalTime

class TimeSlot(
    var id: Int,

    var starts: LocalTime,
    var ends: LocalTime,
    var maxParticipants: Int,
    var participation: Participation,
    var pupils: List<Pupil>
) {

}
